### PR TITLE
[MAINT] `LooseVersion` deprecations

### LINKED ISSUE #3108: distutils.version.LooseVersion is deprecated
distutils is deprecated in python 3.10 (the deprecation warning for `LooseVersion` is showing up in our python 3.10 CIs). Over at MNE-Python we use pkg_resources now instead: 

https://github.com/mne-tools/mne-python/blob/433a4e19bb15b061c17a3fee3e9f8e8f6d8ea229/mne/fixes.py#L50-L55

Statsmodels recently addressed it with `packaging.Version` and `packaging.parse`, which seems just as good: https://github.com/statsmodels/statsmodels/pull/7946/files

### PR DESCRIPTION
**Pull request by NicolasGensollen**
Closes #3108 

Thanks to MNE folks for the fix. :+1: 
I'll add some unit tests for `_compare_version` once the CI is green again.

### CODE REVIEW

File: nilearn/version.py [hunk 1]
```diff
@@ -102,6 +100,42 @@ def _import_module_with_version_check(
     return module
 
 
+def _compare_version(version_a, operator, version_b):
```
Comment (jeromedockes) [inline nilearn/version.py:103]: if this is copied from mne could we add a link to the place it comes from?

File: nilearn/version.py [hunk 2]
```diff
@@ -102,6 +100,42 @@ def _import_module_with_version_check(
...
+    result : :obj:`bool`
+        The result of the version comparison.
+
+    """
+    try:
+        from pkg_resources import parse_version as parse  # noqa:F401
+    except ImportError:
+        from distutils.version import LooseVersion as parse  # noqa:F401
+    return eval(
+        'parse("{0}") {1} parse("{2}")'.format(version_a, operator, version_b)
```
Comment (jeromedockes) [inline nilearn/version.py:135]: In general using `eval` (esp. on unvalidated input) is not a very good habit, could we use a mapping from strings to functions in the standard library module [`operator`](https://docs.python.org/3/library/operator.html) instead?

Comment (NicolasGensollen) [reply nilearn/version.py]: Good point indeed, thanks @jeromedockes !
I added a mapping in 3c6e888

File: nilearn/version.py [hunk 3]
```diff
@@ -102,6 +100,42 @@ def _import_module_with_version_check(
...
+        Second version string.
+
+    Returns
+    -------
+    result : :obj:`bool`
+        The result of the version comparison.
+
+    """
+    try:
+        from pkg_resources import parse_version as parse  # noqa:F401
```
[the remark this answers is not in the document]
Comment (NicolasGensollen) [reply nilearn/version.py:144]: I can add a comment if you think that it could be useful

**Review by jeromedockes — APPROVED** [review]
LGTM, thanks @NicolasGensollen !

File: nilearn/version.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (bthirion) [reply nilearn/version.py:144]: Would be useful indeed.

**Review by bthirion — APPROVED** [review]
Looks good, thx !

**Comment by NicolasGensollen** [comment]
Thanks! Merging...

**Pull request merged**